elasticdl/python/master/evaluation_service.py

```diff
@@ -13,12 +12,49 @@
 class _EvaluationJob(object):
     """Representation of an evaluation job"""
 
-    def __init__(self, model_version, total_tasks=-1):
+    def __init__(self, metrics_dict, model_version, total_tasks=-1):
+        """
+        Args:
+            metrics_dict: A python dictionary. If model has only one output,
+                `metrics_dict` is a dictionary of `{metric_name: metric}`,
+                i.e. `{"acc": tf.keras.metrics.Accuracy()}`.
+                If model has multiple outputs, `metric_dict` is a dictionary of
+                `{output_name: {metric_name: metric}}`,
+                i.e. `{
+                    "output_a": {"acc": tf.keras.metrics.Accuracy()},
+                    "output_b": {"auc": tf.keras.metrics.AUC()},
+                }`. Note that for model with multiple outputs, each metric
+                only uses one output.
+            model_version: The version of the model to be evaluated.
+            total_tasks: The number of evaluation tasks.
+        """
+
         self.model_version = model_version
         self._total_tasks = total_tasks
         self._completed_tasks = 0
-        self._completed_minibatches = 0
-        self._evaluation_metrics = defaultdict()
+        self._init_metrics_dict(metrics_dict)
+
+    def _init_metrics_dict(self, metrics_dict):
+        if not metrics_dict:
+            raise ValueError(
+                "Doing evaluation with empty evaluation metrics dict."
+            )
+        first_metrics = list(metrics_dict.values())[0]
+        if isinstance(first_metrics, dict):
+            self._model_have_multiple_outputs = True
+            self._metrics_dict = metrics_dict
+        else:
+            # When model has only one output, save it in a dict in order to
+            # keep the same data structure as the `metrics_dict` when model
+            # has multiple outputs.
+            self._model_have_multiple_outputs = False
+            self._metrics_dict = {"output": metrics_dict}
```

Comment:
Move "output" to a var/const since it's also used in line 99 which is easy to get messed up

Reply:
Done.